ospf6d/ospf6_zebra.c

```diff
@@ -229,7 +268,9 @@ static void ospf6_zebra_route_update(int type, struct ospf6_route *request)
 	int nhcount;
 	int ret = 0;
 	struct prefix *dest;
+	struct ospf6_area *oa;
 
+	oa = ospf6_area_lookup_by_area_id(request->path.area_id);
```

Comment:
Getting area here should be based on VRF and area ID.

Reply:
Resolved.